Assignment 04

nrogeau opened the pull request:
<!-- You are about to submit an assignment! -->
<!-- For a title, it's enough to indicate the assignment number, eg, `Assignment 01` -->

### Checklist

- [x] Paste a screenshot of the solution whenever possible (`ctrl+v` or `cmd+v` directly on this textbox works)
- [ ] Add optional notes/remarks to be taken into account for evaluating the submission

![Assignment4](https://user-images.githubusercontent.com/57152105/113168191-bc543a00-9244-11eb-864e-e411d1882a2d.PNG)

nrogeau committed:
Create assigment4.ghx

gonzalocasas commented:
@nrogeau: this is looking great! The one thing I'm missing thou is the last bit of the assignment: storing some of those trajectories (all 3 of them) into a json file.

nrogeau committed:
Added json output

@gonzalocasas Indeed, I also noticed this morning I forgot to include the json output. Here it is!

beverlylytle reviewed (approved):
Looks good!  Only a couple of comments.

From your JSON export:
```
def store_trajectory(trajectory):
    with open(FILE, 'a') as f:
        json.dump(trajectory, f, cls=DataEncoder, indent=2)
    pass
```
The line `pass` is unnecessary.  And from the ghx, it's not clear to me that you included the 'pick' trajectories in your JSON.  

Also, it seems you don't use it, but in the export of configurations, it's written
```
def store_configurations(configurations):
    with open(FILE, 'w') as f:
        json.dump(configurations, f, cls=DataEncoder, indent=2)
    pass

if send_to_json is True:
    for configs in configurations:
        store_configurations(configurations)
```
which has the effect of opening, erasing and rewriting `FILE` as many times as there are elements in `configurations` with only the last one being saved.  Changing `'w'` to `'a'` would result in all the configurations being saved.

nrogeau commented:
Thank you for the feedback!
